pinc/Project.inc

```diff
@@ -979,6 +981,31 @@ class Project
             || that_user_is_proj_facilitator($username));
     }
 
+    public function user_can_delete_nonpage_images($username)
+    {
+        if (is_null($username)) {
+            return false;
+        }
+        // SAs can always delete nonpage images
+        if (that_user_is_a_sitemanager($username)) {
+            return true;
+        }
+
+        // PMs/PFs can only delete nonpage images in New
+        // or P1 Unavailable stages
+        if ($this->can_be_managed_by_user($username)) {
+            if ($this->state == PROJ_NEW || $this->state == PROJ_P1_UNAVAILABLE) {
+                return true;
+            } else {
+                return false;
+            }
+        }
+        // Anyone else can't delete images
+        else {
```

Comment:
*minor* You don't need the else before this return since all code paths would've returned before already.

Reply:
I like that better, thanks. I do plan to do it, but I should wait until tomorrow.